### PR TITLE
AspNetCore.Mvc.Razor* Namespace Summaries

### PR DESCRIPTION
**Pull request by TanayParikh**
For: https://github.com/dotnet/aspnetcore/issues/27286

### CODE REVIEW

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Compilation.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Razor.Compilation">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types used to represent a compiled Razor Page or View.</summary>
+    <remarks>Not intended for use in application code.</remarks>
```
Comment (TanayParikh) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Compilation.xml:4]: Tried to determine usability in application code by contained types, please let me know if these assumptions are incorrect.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Razor.TagHelpers">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains default Razor TagHelpers.</summary>
+    <remarks>
+      For more information, see 
+      <see href="https://docs.microsoft.com/en-US/aspnet/core/mvc/views/tag-helpers/intro">
+      Tag Helpers in ASP.NET Core</see>.
+    </remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml:8]:
```suggestion
    <remarks>
      For more information, see 
      <see href="https://docs.microsoft.com/aspnet/core/mvc/views/tag-helpers/intro">
      Tag Helpers in ASP.NET Core</see>.
    </remarks>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml [hunk 3]
```diff
@@ -1,6 +1,10 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Rendering">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains extensions for HTML Helper &amp; classes to build HTML elements.</summary>
+    <remarks>
+      For more information, see 
+      <see href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.ihtmlhelper">
+      IHtmlHelper Interface</see>.
+    </remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml:8]:
```suggestion
    <remarks>
      For more information, see <see cref="T:Microsoft.AspNetCore.Mvc.Rendering.IHtmlHelper" />.
    </remarks>
```

**Review by JamesNK — APPROVED** [review]

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains classes to access Razor project files &amp; compilation options.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml:3]:
```suggestion
    <summary>Contains classes to access Razor project files and compilation options.</summary>
```
@guardrex  do you understand the `&amp;`

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml [hunk 5]
```diff
@@ -1,6 +1,8 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Rendering">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains extensions for HTML Helper &amp; classes to build HTML elements.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml:3]:
```suggestion
    <summary>Contains extensions for HTML Helper and classes to build HTML elements.</summary>
```

**Review by Rick-Anderson — APPROVED** [review]
I don't understand the `&amp;`

**Comment by TanayParikh** [comment]
Is this good to merge now? @pranavkm any concerns?

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.Razor">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types used to represent &amp; render a Razor Page or View.</summary>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.xml:3]:
```suggestion
    <summary>Contains types used to represent and render a Razor Page or View.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.xml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types to handle &amp; execute a Razor Page.</summary>
```
Comment (JamesNK) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.xml:3]:
```suggestion
    <summary>Contains types to handle and execute a Razor Page.</summary>
```

**Review by JamesNK — APPROVED** [review]

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[35f6bbf](https://github.com/dotnet/AspNetApiDocs/commits/35f6bbfcf55e996d1b55ece1cde2b33a9657c6a7)_: 

:clock10: **Pending**: waiting for processors ([12](https://ops.microsoft.com/#/statistic) builds ahead of you) 

 :warning: **Docs Build is busy**, currently there are 12 builds ahead of this one, for more information you can view the Build queue graph on the [Docs Portal](https://ops.microsoft.com/#/statistic).

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[35f6bbf](https://github.com/dotnet/AspNetApiDocs/commits/35f6bbfcf55e996d1b55ece1cde2b33a9657c6a7)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-0f020693)

<a id="0f020693"></a>
###
 - **[Error-RuningBuildFailed]** `````Some unexpected errors happened when running build, please open a ticket in https://SiteHelp and include the error report for our team to troubleshoot`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C4%5Cbaf25683-9e81-c379-8ba7-62f30178bab2%5CPullRequest%5C202012040119293872-179%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=jHTJLfFGmhkWdGQKiRTVyVQiNRGU766FSiL84331CPQ%3D&st=2020-12-04T01%3A15%3A46Z&se=2021-01-04T01%3A20%3A46Z&sp=r).

If you see build warnings/errors with permission issues, it might be due to single sign-on (SSO) enabled on Microsoft's GitHub organizations. Please follow instructions [here](https://teams.microsoft.com/l/message/19:7ecffca1166a4a3986fed528cf0870ee@thread.skype/1590030602688?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&parentMessageId=1590030602688&teamName=Docs%20Support&channelName=General&createdTime=1590030602688) to re-authorize your GitHub account to Docs Build.

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/baf25683-9e81-c379-8ba7-62f30178bab2?tabName=brokenlinks).

**Note:** Your PR may contain errors or warnings unrelated to the files you changed. This happens when external dependencies like GitHub alias, Microsoft alias, cross repo links are updated. Please use these [instructions](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master&tabs=pr-validation-error-solution#q-why-do-i-get-pr-validation-result-with-cryptic-error-or-warning-messages-not-related-to-the-changes-i-make-in-my-working-branch) to resolve them.

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[35f6bbf](https://github.com/dotnet/AspNetApiDocs/commits/35f6bbfcf55e996d1b55ece1cde2b33a9657c6a7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Compilation.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Compilation.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.compilation?view=aspnetcore-1.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Infrastructure.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.Infrastructure.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.infrastructure?view=aspnetcore-2.2&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.runtimecompilation?view=aspnetcore-3.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.taghelpers?view=aspnetcore-1.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor?view=aspnetcore-1.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razorpages.infrastructure?view=aspnetcore-2.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.RazorPages.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razorpages?view=aspnetcore-2.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering?view=aspnetcore-1.0&branch=pr-en-us-179) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C4%5Cbaf25683-9e81-c379-8ba7-62f30178bab2%5CPullRequest%5C202012041416042494-179%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=kZwbzoAM3J0zpqyqwk%2BnB%2FQO6G6pf6LbTEL%2F%2BwuzCPU%3D&st=2020-12-04T14%3A15%3A29Z&se=2021-01-04T14%3A20%3A29Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/baf25683-9e81-c379-8ba7-62f30178bab2?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Review by pranavkm — APPROVED** [review]

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[62dad8b](https://github.com/dotnet/AspNetApiDocs/commits/62dad8b238ed262a55cf7d6bcef3b2cf5ff725ed)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml) | :x:Error | | [Details](#user-content-13d9fb1d-c8876c1a)

<a id="13d9fb1d-c8876c1a"></a>
### [aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml)
 - **Line 0, Column 0**: **[Error-ECMA2Yaml_InternalError]** 
`````
Intenal Several Error: System.Xml.XmlException: 'Tag' is an unexpected token. Expecting whitespace. Line 4, position 109.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.Throw(String res, String arg)
   at System.Xml.XmlTextReaderImpl.ParseAttributes()
   at System.Xml.XmlTextReaderImpl.ParseElement()
   at System.Xml.XmlTextReaderImpl.ParseElementContent()
   at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r)
   at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r, LoadOptions o)
   at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XDocument.Parse(String text, LoadOptions options)
   at ECMA2Yaml.ECMALoader.LoadType(FileItem typeFile)
   at ECMA2Yaml.ECMALoader.LoadTypes(String basePath, Namespace ns)
`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C4%5Cbaf25683-9e81-c379-8ba7-62f30178bab2%5CPullRequest%5C202012041428388436-179%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=YKkb%2BkPhX3QyECvC6WdLYjNeiCYlJbycjMALuCWUHq4%3D&st=2020-12-04T14%3A24%3A23Z&se=2021-01-04T14%3A29%3A23Z&sp=r).

If you see build warnings/errors with permission issues, it might be due to single sign-on (SSO) enabled on Microsoft's GitHub organizations. Please follow instructions [here](https://teams.microsoft.com/l/message/19:7ecffca1166a4a3986fed528cf0870ee@thread.skype/1590030602688?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&parentMessageId=1590030602688&teamName=Docs%20Support&channelName=General&createdTime=1590030602688) to re-authorize your GitHub account to Docs Build.

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/baf25683-9e81-c379-8ba7-62f30178bab2?tabName=brokenlinks).

**Note:** Your PR may contain errors or warnings unrelated to the files you changed. This happens when external dependencies like GitHub alias, Microsoft alias, cross repo links are updated. Please use these [instructions](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master&tabs=pr-validation-error-solution#q-why-do-i-get-pr-validation-result-with-cryptic-error-or-warning-messages-not-related-to-the-changes-i-make-in-my-working-branch) to resolve them.

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Comment by scottaddie** [comment]
Docs Build status updates of commit _[fc771da](https://github.com/dotnet/AspNetApiDocs/commits/fc771da06c5f81c63e7ba1867c10616ac65330d3)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.runtimecompilation?view=aspnetcore-3.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Razor.TagHelpers.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.razor.taghelpers?view=aspnetcore-1.0&branch=pr-en-us-179) |
[aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml](https://github.com/dotnet/AspNetApiDocs/blob/taparik/razorNamespaceSummaries/aspnet-core/xml/ns-Microsoft.AspNetCore.Mvc.Rendering.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering?view=aspnetcore-1.0&branch=pr-en-us-179) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C4%5Cbaf25683-9e81-c379-8ba7-62f30178bab2%5CPullRequest%5C202012041430594262-179%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=S2uOiOM8U9W8%2FrQpA2P%2FjCOzxk0GB2oI6H8FbjoQ%2FPs%3D&st=2020-12-04T14%3A29%3A53Z&se=2021-01-04T14%3A34%3A53Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/baf25683-9e81-c379-8ba7-62f30178bab2?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Pull request merged**